uniffi_bindgen/src/interface/attributes.rs

```diff
@@ -315,6 +313,78 @@ impl TryFrom<&weedle::attribute::ExtendedAttributeList<'_>> for ConstructorAttri
     }
 }
 
+/// Represents UDL attributes that might appear on a method.
+///
+/// This supports the `[Throws=ErrorName]` attribute for methods that can produce
+/// an error, and the `[Self=ByArc]` attribute for methods that take `Arc<Self>` as receiver.
+#[derive(Debug, Clone, Hash, Default)]
+pub(super) struct MethodAttributes(Vec<Attribute>);
+
+impl MethodAttributes {
+    pub(super) fn get_throws_err(&self) -> Option<&str> {
+        self.0.iter().find_map(|attr| match attr {
+            // This will hopefully return a helpful compilation error
+            // if the error is not defined.
+            Attribute::Throws(inner) => Some(inner.as_ref()),
+            _ => None,
+        })
+    }
+
+    pub(super) fn get_by_arc(&self) -> bool {
+        self.0
+            .iter()
+            .any(|attr| matches!(attr, Attribute::SelfType(SelfType::ByArc)))
+    }
+}
+
+impl TryFrom<&weedle::attribute::ExtendedAttributeList<'_>> for MethodAttributes {
+    type Error = anyhow::Error;
+    fn try_from(
+        weedle_attributes: &weedle::attribute::ExtendedAttributeList<'_>,
+    ) -> Result<Self, Self::Error> {
+        let attrs = parse_attributes(weedle_attributes, |attr| match attr {
+            Attribute::SelfType(_) => Ok(()),
+            Attribute::Throws(_) => Ok(()),
+            _ => bail!(format!("{:?} not supported for methods", attr)),
+        })?;
+        Ok(Self(attrs))
+    }
+}
+
+impl<T: TryInto<MethodAttributes, Error = anyhow::Error>> TryFrom<Option<T>> for MethodAttributes {
+    type Error = anyhow::Error;
+    fn try_from(value: Option<T>) -> Result<Self, Self::Error> {
+        match value {
+            None => Ok(Default::default()),
+            Some(v) => v.try_into(),
+        }
+    }
+}
+
+/// Represents the different possible types of method call receiver.
+///
+#[derive(Debug, Clone, Hash)]
+pub(super) enum SelfType {
+    ByArc, // Method receiver is `Arc<Self>`.
+    ByRef, // Method receiver is `&self`.
```

Comment:
I don't feel too strongly about this, but I fear supporting `ByRef` here might just cause confusion. IIRC, there will never be a reason to specify it, which will confuse when someone specifies it anyway. Then some people will specify it everywhere as a convention, then get confused when *not* specifying doesn't mean the receiver consumes self (ie, they might expect support for  `self: Self`)

Reply:
That's fair; I'll remove it before merge, it's easy to bring back later if we need it.